src/docs/product.yaml

```diff
@@ -0,0 +1,89 @@
+paths:
+  /api/post:
+    post:
+      tags:
+        - CRUD posts
+      descriptions: 'Creating a new post'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                title:
+                  type: string,
+                description:
+                  type: string,
+                photos:
+                  type: array,
+                category:
+                  type: string,
+                location:
+                  type: string,
+                productCondition:
+                  type: string,
+                shippingOptions:
+                  type: string,
+                postType:
+                  type: string,
+                donor:
+                  type: string,
+                beneficiary:
+                  type: string,
+                isEvent:
+                  type: boolean
+            examples:
+              1:
+                value: '{
+                  "title": "Red couch",
+                  "description": "I have an underused couch I don''t need anymore",
+                  "photos": ["redCouch0.jpg", "redCouch1.jpg"],
+                  "category": "Furniture",
+                  "location": "Istanbul, Kadikoy",
+                  "productCondition": "Underused",
+                  "shippingOptions": "Meet up",
+                  "postType": "Donate",
+                  "isEvent": false
+                  }'
+      responses:
+        '201':
+          description: 'Created (post)'
+        '400':
+          description: 'Bad request (either validation or database error)'
+        '500':
+          description: 'Internal server error'
+components:
+  schemas:
+    Post:
+      type: object
+      required:
+      properties:
```

Comment:
i think we should add a small description after all properties of both models but if it takes too much time, maybe we can leave this job after everything is done

Reply:
We might add descriptions, but tbh most of 'em seems self-explanatory. I only added to the ones that _might_ be confusing.